From 8557469a6198edb9194265513fde9a10e0d87d41 Mon Sep 17 00:00:00 2001
From: Lars Kellogg-Stedman <lars@redhat.com>
Date: Mon, 2 Nov 2020 13:38:47 -0500
Subject: [PATCH] handle missing access_rules

we delete the access_rules for keystone_version < 3.13, but we don't
make the same check in the cleanup method, leading to a KeyError
exception.

Change-Id: I02e124d90f99d400d8c59bff2c563fdc85e624d4
---
 .../dashboards/identity/application_credentials/forms.py        | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/openstack_dashboard/dashboards/identity/application_credentials/forms.py b/openstack_dashboard/dashboards/identity/application_credentials/forms.py
index c57132b40..c63462d05 100644
--- a/openstack_dashboard/dashboards/identity/application_credentials/forms.py
+++ b/openstack_dashboard/dashboards/identity/application_credentials/forms.py
@@ -138,6 +138,8 @@ class CreateApplicationCredentialForm(forms.SelfHandlingForm):
         except yaml.YAMLError:
             msg = (_('Access rules must be a valid JSON or YAML list.'))
             raise forms.ValidationError(msg)
+        except KeyError:
+            pass
         return cleaned_data
 
 
-- 
2.25.4

